test/modules/encoders/test_clip_text_encoder.py

```diff
@@ -0,0 +1,93 @@
+
+        text_encoder = CLIPTextEncoder(embedding_dim=embedding_dim, use_clip_init=True)
+        text_encoder = text_encoder.to(device)
+        assert isinstance(text_encoder, torch.nn.Module)
+
+        out = text_encoder(text)
+        assert out.shape == (batch_size, embedding_dim)
+        assert_expected(out.sum().item(), -143.8726, rtol=0, atol=1e-3)
+
+    def test_no_init(self):
+        embedding_dim = 256
+        batch_size = 16
+        text = torch.randint(1, 79, (batch_size, context_length), dtype=torch.long)
+
+        text_encoder = CLIPTextEncoder(embedding_dim=embedding_dim, use_clip_init=False)
+        text_encoder = text_encoder.to(device)
+        assert isinstance(text_encoder, torch.nn.Module)
+
+        out = text_encoder(text)
+        assert out.shape == (batch_size, embedding_dim)
+        assert_expected(out.sum().item(), 64.1976, rtol=0, atol=1e-3)
+
+    def test_small_embedding(self):
+        embedding_dim = 1
+        batch_size = 16
+        text = torch.randint(1, 79, (batch_size, context_length), dtype=torch.long)
+
+        text_encoder = CLIPTextEncoder(embedding_dim=embedding_dim, use_clip_init=True)
+        text_encoder = text_encoder.to(device)
+        assert isinstance(text_encoder, torch.nn.Module)
+
+        out = text_encoder(text)
+        assert out.shape == (batch_size, embedding_dim)
+        assert_expected(out.sum().item(), -1.4961, rtol=0, atol=1e-3)
+
+    def test_small_batch(self):
+        embedding_dim = 256
+        text = torch.randint(1, 79, (context_length,), dtype=torch.long).unsqueeze(0)
+
+        text_encoder = CLIPTextEncoder(embedding_dim=embedding_dim, use_clip_init=True)
+        text_encoder = text_encoder.to(device)
+        assert isinstance(text_encoder, torch.nn.Module)
+
+        out = text_encoder(text)
+        assert out.shape == (1, embedding_dim)
+        assert_expected(out.sum().item(), 3.9050, rtol=0, atol=1e-3)
+
+    def test_under_context_length(self):
+        embedding_dim = 256
+        batch_size = 16
+        text = torch.randint(1, 79, (batch_size, context_length - 5), dtype=torch.long)
+
+        text_encoder = CLIPTextEncoder(embedding_dim=embedding_dim, use_clip_init=True)
+        text_encoder = text_encoder.to(device)
+        assert isinstance(text_encoder, torch.nn.Module)
+
+        out = text_encoder(text)
+        assert out.shape == (batch_size, embedding_dim)
+        assert_expected(out.sum().item(), -142.1651, rtol=0, atol=1e-3)
+
+    def test_over_context_length(self):
+        embedding_dim = 256
+        batch_size = 16
+        text = torch.randint(1, 79, (batch_size, context_length + 5), dtype=torch.long)
+
+        text_encoder = CLIPTextEncoder(embedding_dim=embedding_dim, use_clip_init=True)
+        text_encoder = text_encoder.to(device)
+        assert isinstance(text_encoder, torch.nn.Module)
+
+        with pytest.raises(IndexError):
+            out = text_encoder(text)
```

Comment:
You can also add a test that the module can be torchscripted (I think it should be able to be scripted, if not let me know). [Example](https://github.com/facebookresearch/multimodal/blob/main/test/modules/encoders/test_mil_encoder.py#L76)

Reply:
Added a scripting test but also modified CLIPTextEncoder to pass the test -- see recent changes